pkg/resources/cloudcontroller/vsphere.go

```diff
@@ -145,7 +145,7 @@ func getVsphereCPIVersion(version semver.Semver) (string, error) {
 	case 21:
 		return latestVsphereCPIVersion, nil
 	case 22:
-		return latestVsphereCPIVersion, nil
+		return "", fmt.Errorf("kubernetes v1.22 not supported by external CCM")
```

Comment:
This is kind of a conflict for me. One the one hand, we make it configurable what is supported and what not, but this hardcodes it again to the KKP codebase. So which wins?

Can we rewrite this to make use of the configured incompatibilities? WDYT @xmudrii ? Technically this then also, in the future, affects other providers where we currently have hardcoded `switch` statements that technically would need to be moved to this new configuration file, right?

Reply:
I agree that having this hardcoded switch is in conflict with the goal of this PR. We can use the incompatibilities file also to check these kinds of conditions without too much effort if it worths it.

Reply:
Yes, I agree as well.

Reply:
@xmudrii @xrstf I used the incompatibilities mechanism to block also the migration, this ad-hoc part for vSphere has been deleted.